Packs/Workday/Integrations/WorkdayIAMEventsGenerator/WorkdayIAMEventsGenerator.py

```diff
@@ -478,585 +481,215 @@
+    existing_email_addressees = []
+    for report in integration_context['Report_Entry']:
+        email_address = report.get('Email_Address')
+        existing_email_addressees.append(email_address)
+    if user_email not in existing_email_addressees:
+        raise Exception(f'The user email {user_email} does not exist. Please try one  of the followings: '
+                        f'ttang@test.com, rrahardjo@test.com, sarnold@test.com')
+
     else:
+        # Get the specific report and change the relevant field for rehire event.
+        user_report = [i for i in integration_context['Report_Entry'] if (i['Email_Address'] == user_email)][0]
+        is_terminated = user_report.get('Employment_Status')
+        rehired_status = user_report.get('Rehired_Employee')
+        if user_email in existing_email_addressees and is_terminated == 'Active' or rehired_status == 'Yes':
+            raise Exception(f'The user {user_email} is not terminated. Either he is still active or was already '
+                            f'rehired.')
+        user_report['Rehired_Employee'] = 'Yes'
+        user_report['Prehire_Flag'] = 'True'
+        set_integration_context(integration_context)
+        return_results('Successfully generated the rehire user event.')
+
+
+def main():
+
+    if demisto.command() == 'test-module':
+        test_module()
+
+    elif demisto.command() == 'long-running-execution':
         integration_context = get_integration_context()
-        reports = REHIRE_USER_REPORT['Report_Entry']
-        for report in reports:
-            email = report.get('Email_Address')
-            if email == user_email:
-                users = [i for i in integration_context['Report_Entry'] if not (i['Email_Address'] == email)]
-                users.append(report)
-                integration_context['Report_Entry'] = users
-                set_integration_context(integration_context)
+        if not integration_context:
+            set_integration_context(FIRST_RUN_REPORT)
+        while True:
+            port = int(demisto.params().get('longRunningPort', ''))
+            server = WSGIServer(('0.0.0.0', port), APP)
+            server.serve_forever()
 
+    elif demisto.command() == 'workday-generate-hire-event':
+        get_new_hire_reports()
 
-if demisto.command() == 'long-running-execution':
-    integration_context = get_integration_context()
-    if not integration_context:
-        set_integration_context(FIRST_RUN_REPORT)
-    while True:
-        port = int(demisto.params().get('longRunningPort', ''))
-        server = WSGIServer(('0.0.0.0', port), APP)
-        server.serve_forever()
+    elif demisto.command() == 'workday-generate-update-event':
+        get_update_report()
 
+    elif demisto.command() == 'workday-generate-rehire-event':
+        get_rehire_report()
 
-elif demisto.command() == 'generate-event':
-    generate_event()
+    elif demisto.command() == 'workday-generate-terminate-event':
+        get_terminate_report()
```

Comment:
replace the function names to be `generate_*` and not `get_*` , because we generate terminate report

Reply:
same for other functions